### PR TITLE
Save comment_type for product reviews

### PR DESCRIPTION
**Pull request by claudiosanches**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Reviews are saved with incorrect `comment_type` into the database, this PR solves it.

Note: We should think about some upgrade routine to fix all old reviews. cc @claudiulodro @rodrigoprimo 

### How to test the changes in this Pull Request:

1. Leave a review for any product, note that this need to be done in the store's front.
2. Check database for `comment_type` column, now shows `review` instead of blank.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### Changelog entry

> Dev - Fixed how save comment type for product reviews.

### CODE REVIEW

File: includes/class-wc-comments.php [hunk 1]
```diff
@@ -405,6 +408,21 @@ public static function get_rating_counts_for_product( &$product ) {
...
+
+	/**
+	 * Update comment type of product reviews.
+	 *
+	 * @since 3.5.0
+	 * @param array $comment_data Comment data.
+	 * @return array
+	 */
+	public static function update_comment_type( $comment_data ) {
+		if ( ! is_admin() && isset( $_POST['comment_post_ID'] ) && 'product' === get_post_type( absint( $_POST['comment_post_ID'] ) ) ) { // WPCS: input var ok, CSRF ok.
```
Comment (claudiulodro) [inline includes/class-wc-comments.php]: This should probably check to make sure `'' === comment_data['comment_type']` so we don't overwrite some other type of product comments.

**Review by claudiulodro — DISMISSED** [review]
Good catch. I've confirmed this solves the issue. Let's add the update routine as part of this PR (we can merge this after beta is tagged. API is priority number 1 right now).

The tricky part with the update routine is figuring out which comments to update. We probably only want to update comments that have a product parent and have `''` as the comment type.

**Comment by claudiosanches** [comment]
@claudiulodro I included to check for empty `comment_type`, now just missing the upgrade routine.

**Comment by willgorham** [comment]
I had a faint memory of this coming up before so I did a quick issue search: #20247. I'm not sure if making this change will actually affect anything that Mike mentioned, but I figured I'd drop this here FYI. If this isn't a concern, ignore me 😄

**Comment by claudiulodro** [comment]
Ah! Thanks for the heads up @willgorham!

Reading through that it looks like no comment_type is intentional and we should not make this change @claudiosanches. We should document this somehow.

**Comment by claudiulodro** [comment]
I know! A unit test. We should definitely start documenting this on-purpose counterintuitive stuff with unit tests. Something like

```
/**
 * Reviews should not have a comment type for maximum compatibility.
*/
function testReviewCommentType() {
    ...
    $review_comment_type = $review->comment_type;
    $this->assertEquals( '', $review_comment_type );
}
```

I'll open a PR in the morning.

**Comment by claudiosanches** [comment]
>To me, it looks like setting the comment_type to review doesn't have any negative impact. Reviews are still included in the default admin comment interface (list table, and on single products), still exported as a comment in GDPR tools, still treated as regular old comments... The only thing that is different is the display in the admin dashboard 'Activity' widget, which shows the comment as a 'Review', but otherwise normal.

I agree with that, there is no negative impact, also make easy to query reviews anytime without a JOIN.
And as Justin said, there is already places in our code base that already uses `review`, like in the REST API.

**Comment by claudiulodro** [comment]
I've added an update routine to set the 'review' type for comments that don't have it. It identifies the appropriate comments by looking for comments that don't have a comment type and have a product parent.

Easiest way to test is to add this code somewhere and refresh the page:
```
require_once( 'includes/wc-update-functions.php' );
add_action( 'init', 'wc_update_350_reviews_comment_type' );
```

**Review by kloon — APPROVED** [review]
Changes looks good to me now!

**Pull request merged**